src/libostree/ostree-repo-refs.c

```diff
@@ -107,7 +107,38 @@ write_checksum_file_at (OstreeRepo   *self,
 
     if (!_ostree_repo_file_replace_contents (self, dfd, name, (guint8*)bufnl, l + 1,
                                              cancellable, error))
-      goto out;
+      {
+        if (g_error_matches (*error, G_IO_ERROR, G_IO_ERROR_IS_DIRECTORY))
```

Comment:
The GError conventions allow the caller to pass `NULL`.  So anytime we want to do the equivalent of `try/catch`, we need to do:

```
g_autoptr(GError) temp_error = NULL;
if (!somefunction (..., &temp_error)) {
   if (g_error_matches (temp_error, ...)) {
     /* Handle error */
   } else {
     g_propagate_error (error, g_steal_pointer (&temp_error));
   }
}